### PR TITLE
Hot Strip algorithm updated

### PR DESCRIPTION
**Pull request by suchandradutta**
Updating Hot Strip algorithms in CalibTracker/SiStripQuality putting extra requirement on the strip Digi wrt APV median as developed by Mario Galanti + Message logger ourput made more logical in DQMOffline/CalibTracker/plugins/SiStripQualityHotStripIdentifierRoot.cc

### CODE REVIEW

**Comment by mmusich** [comment]
please test

**Comment by deguio** [comment]
+1

File: DQMOffline/CalibTracker/plugins/SiStripQualityHotStripIdentifierRoot.cc [hunk 1]
```diff
@@ -201,7 +201,7 @@ SiStripBadStrip* SiStripQualityHotStripIdentifierRoot::getNewObject(){
 void SiStripQualityHotStripIdentifierRoot::algoBeginRun(const edm::Run& iRun,const edm::EventSetup& iSetup){
   //Retrieve tracker topology from geometry
   edm::ESHandle<TrackerTopology> tTopoHandle;
-  iSetup.get<TrackerTopologyRcd>().get(tTopoHandle);
+  iSetup.get<IdealGeometryRecord>().get(tTopoHandle);
   tTopo = tTopoHandle.product();
```
Comment (mmusich) [inline DQMOffline/CalibTracker/plugins/SiStripQualityHotStripIdentifierRoot.cc]: @suchandradutta why this changes? I dont' see them in the corresponding commit in the 7.6.X branch b33a00a845324e8b544f79fee4f3db89d91aa423. I suspect this causes the crash the test observe.

**Comment by mmusich** [comment]
@cmsbuild please aknowledge changes

**Comment by mmusich** [comment]
@cmsbuild please test

**Comment by smuzaffar** [comment]
@mmusich , cms-bot will acknowledge it as soon as it finds some resources in jenkins :-(

**Comment by mmusich** [comment]
@smuzaffar sorry, one gets used to see it working on the spot. Is there any intervention on the system ongoing? I have noticed resources tend to be scarce these days.

**Comment by deguio** [comment]
+1

**Comment by mmusich** [comment]
+1 
for https://github.com/suchandradutta/cmssw/commit/eac661f2484a63dac2713052e2199bbf9f46eed4 (backport of #9982)
Visual inspection of the code w.r.t 76X changes OK

**Comment by davidlange6** [comment]
+1
